### PR TITLE
Create qmcio_hdf and qmcio_xml cmake targets

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Create qmcio_hdf and qmcio_xml cmake targets by splitting qmcutils which is overcrowded.
qmcio_hdf is for source files previously at folder src/io. Files are now moved to src/io/hdf
qmcio_xml is for source files previously at folder src/OhmmsData. Files are now moved to src/io/xml
qmcio target is just an interface target including both qmcio_hdf and qmcio_xml.

Additional cleaning at src/Platforms
Platforms/accelerators.cpp has been moved to QMCApp because it is part of the app logic not the platforms building block.
tests in src/Platforms/XXX/tests are moved to src/Platforms/tests/XXX.

## What type(s) of changes does this code introduce?
- Refactoring (no functional changes, no api changes)
- Build related changes

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
Bora

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: src/Platforms/tests/CPU/test_e2iphi.cpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 #include <stdio.h>
 #include <string>
-#include "e2iphi.h"
+#include <CPU/e2iphi.h>
```
Comment (prckent) [inline src/Platforms/tests/CPU/test_e2iphi.cpp:17]: Quotes here?

File: src/Platforms/tests/OpenMP/test_class_member.cpp [hunk 2]
```diff
@@ -15,8 +15,8 @@
 #include <memory>
 #include <vector>
 #include <iostream>
-#include "omp.h"
-#include "OMPallocator.hpp"
+#include <omp.h>
+#include <OpenMP/OMPallocator.hpp>
```
Comment (prckent) [inline src/Platforms/tests/OpenMP/test_class_member.cpp:19]: Quotes again?

File: src/QMCDrivers/CloneManager.cpp [hunk 3]
```diff
@@ -21,7 +21,7 @@
 #include "Message/Communicate.h"
 #include "Message/OpenMP.h"
 #include "Utilities/IteratorUtility.h"
-#include <qmc_common.h>
+#include <Utilities/qmc_common.h>
```
Comment (prckent) [inline src/QMCDrivers/CloneManager.cpp:24]: Quotes again? How is this different from the usage in qmcapp.cpp

**Review by prckent — COMMENTED** [review]
Seems like a good direction.

A number of our includes are now included via <dir/file.h> and not "dir/file.h". What is your logic?

File: src/Platforms/tests/CPU/test_e2iphi.cpp (hunk 1 above)
Comment (ye-luo) [reply src/Platforms/tests/CPU/test_e2iphi.cpp:17]: Will be fixed in a separate PR.

File: src/QMCDrivers/CloneManager.cpp (hunk 3 above)
Comment (ye-luo) [reply src/QMCDrivers/CloneManager.cpp:24]: Will be fixed in a separate PR.

File: src/Platforms/tests/OpenMP/test_class_member.cpp (hunk 2 above)
Comment (ye-luo) [reply src/Platforms/tests/OpenMP/test_class_member.cpp]: `omp.h` is from compiler or OS. So use angle bracket.
`OpenMP/OMPallocator.hpp` is from QMCPACK, so should be quotes.
I fixed this place in this PR but need to fix other files in the repo using a script in a separate PR.

**Review by prckent — APPROVED** [review]
OK, good to further improve in a later PR. No need to iterate on this one.

**Pull request merged**